### PR TITLE
deno: 1.36.0 -> 1.37.1

### PR DESCRIPTION
**Pull request by 06kellyjac**
## Description of changes

Update deno to `1.37.1`
Also fix update script so it can find the v8 crate dependency version

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/web/deno/update/librusty_v8.ts [hunk 1]
```diff
@@ -74,7 +68,10 @@ export async function updateLibrustyV8(
 ) {
   log("Starting librusty_v8 update");
   // 0.0.0
-  const version = await getLibrustyV8Version(owner, repo, denoVersion);
+  const cargoLockData = await getCargoLock(owner, repo, denoVersion);
+  console.log(cargoLockData);
+  const packageItem = cargoLockData.package.find(({ name }) => name === "v8");
```
Comment (marsam) [inline pkgs/development/web/deno/update/librusty_v8.ts:73]: Would this resolve https://github.com/NixOS/nixpkgs/pull/256753?

Comment (06kellyjac) [reply pkgs/development/web/deno/update/librusty_v8.ts:73]: Yes it would. Sorry I missed this @zvxzvxzv and thanks for the PR

**Comment by meatcar** [comment]
Result of `nixpkgs-review pr 257617` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>deno</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
    <li>windmill</li>
  </ul>
</details>

**Comment by meatcar** [comment]
Trying to get this PR to build deno,

```nix
nativeBuildInputs = [
  installShellFiles
  cmake # for libz-ng-sys crate
];
PROTOC = "${protobuf}/bin/protoc"; # for deno_kv crate
```

I hit an error that's outside my expertise:
```
   ...
   Compiling deno v1.37.1 (/build/source/cli)
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', cargo-auditable/src/rustc_wrapper.rs:91:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: could not compile `deno` (bin "deno")
```

**Comment by fabianhjr** [comment]
Result of `nixpkgs-review pr 257617` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>deno</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
    <li>windmill</li>
  </ul>
</details>

**Comment by 06kellyjac** [comment]
interesting. my fix wasn't much different but worked fine for me. if you could try build this again that'd be great @meatcar

**Comment by meatcar** [comment]
Result of `nixpkgs-review pr 257617` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>deno</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>vimPlugins.denops-vim</li>
    <li>vimPlugins.vim-pluto</li>
    <li>windmill</li>
  </ul>
</details>

**Comment by meatcar** [comment]
Strange, it builds successfully today.

**Review by marsam — APPROVED** [review]

**Pull request merged**